### PR TITLE
FIPS: Add support for starting VM in FIPS mode.

### LINKED ISSUE #2170: Add support for FIPS mode
Reference: https://github.com/kata-containers/packaging/issues/787

### PR DESCRIPTION
**Pull request by amshinde**
FIPS are a set of security standards for encryption algororithms
in user and kernel space among others.
Have Kata support this by starting the VM for a container
in FIPS mode on detecting that the host is running in FIPS mode.

Depends-on: github.com/kata-containers/packaging#788

Fixes #2170

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

**Comment by amshinde** [comment]
Depends on https://github.com/kata-containers/packaging/pull/788

File: pkg/katautils/create.go [hunk 1]
```diff
@@ -175,6 +182,35 @@ func CreateSandbox(ctx context.Context, vci vc.VC, ociSpec specs.Spec, runtimeCo
...
+func checkForFIPS(sandboxConfig *vc.SandboxConfig) error {
+	content, err := ioutil.ReadFile(procFIPS)
+	if err != nil {
+		// In case file cannot be found or read, simply return
+		return nil
+	}
+
+	enabled, err := strconv.Atoi(strings.Trim(string(content), "\n\t "))
+	if err != nil {
+		// Enexpected format, ignore and simply return early
```
Comment (jodh-intel) [inline pkg/katautils/create.go]: Nit: typo - "unexpected".

**Comment by devimc** [comment]
commit message: typo: s/algororithms/algorithms

**Comment by amshinde** [comment]
Typos fixed.
/test

**Comment by bergwolf** [comment]
failed ci looks unrelated (https://github.com/kata-containers/tests/issues/2069)
/merging...

**Pull request merged**